plugins/git/neogit/default.nix

```diff
@@ -122,8 +128,9 @@ helpers.neovim-plugin.mkNeovimPlugin config {
           "fzf-lua"
         ];
 
-    extraPackages = optional (hasInfix "which" (
-      cfg.settings.commit_view.verify_commit.__raw or ""
-    )) pkgs.which;
+    extraPackages = [
+      cfg.gitPackage
+    ] ++ optional (hasInfix "which" (cfg.settings.commit_view.verify_commit.__raw or "")) pkgs.which;
```

Comment:
I wonder if for neogit we need a more freeform approach, e.g. `attrsOf (nullOr package)`.

Or at least a `whichPackage` option.

Saying that, is it even possible to do anything without the `which` command? Installing the dependency seems a little overkill. Maybe it's for cross-platform reasons 🤷

Reply:
I have a whichPackage in https://github.com/nix-community/nixvim/pull/2114